File: Tower of Hanoi/TowerOfHanoi.java
import java.util.*;

public class TowerOfHanoi
{
    static int moves = 0;
    static void towerOfHanoi(int n, String source, String dest, String temp)
    {
        if (n == 1)
        {
            System.out.println("Moving disk " + n + " from " + source + " to " + dest);
            moves++;
        }

        else
        {
            towerOfHanoi(n-1, source, temp, dest);
            System.out.println("Moving disk " + n + " from " + source + " to " + dest);
            moves++;
            towerOfHanoi(n-1, temp, dest, source);
        }
    }

    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter the number of disks you want");
        int numberOfDisks = scan.nextInt();

        towerOfHanoi(numberOfDisks, "A", "B", "C");
        System.out.println("Solved in " + moves + " moves.");

    }

}

File: Tower of Hanoi/tower_of_hanoi.py
# TOWER OF HANOI using recursion

moves = 0
def tower_of_hanoi(n, source, dest, temp):
    
    global moves
    
    if n == 1:
        print("Moving disk ", n, " from ", source, " to ", dest)
        moves += 1
        
    else:
        tower_of_hanoi(n-1, source, temp, dest)
        print("Moving disk ", n, " from ", source, " to ", dest)
        tower_of_hanoi(n-1, temp, dest, source)
        moves += 1
        
n = int(input("Enter the number of disks you want\n"))
print("Minimum number of moves required to complete this: ", (2 ** n) - 1, sep = "", end = "\n\n")

tower_of_hanoi(n, "A", "B", "C")
print("\nCompleted in ", moves, " moves.", sep = "")
